### ISSUE TITLE
Accept or validate baseline index dtypes in JAX FFI operators

### ISSUE THREAD

**Issue by chrisfinlay** [epfl-radio-astro/ri-kernels#4]
## Problem

`RFIVisOp` on `main` preserves the dtype of `a1` and `a2` when constructing all baseline metadata, while the native FFI handlers require `S32` operands. With `jax_enable_x64=True`, ordinary arrays such as `jnp.array([0])` default to `int64`, and evaluation fails at runtime with:

```
INVALID_ARGUMENT: Failed to decode all FFI handler operands
Wrong buffer dtype: expected S32 but got S64
```

The new `RFIDelayVisOp` added by PR #3 inherits the same behavior. This is especially easy to encounter in the advertised float64/complex128 configuration because JAX x64 must be enabled.

The current tests avoid the failure by explicitly constructing baseline pairs with `dtype=np.int32`.

## Suggested resolution

Either:

- normalize `a1`, `a2`, and all derived sorter/start arrays to `jnp.int32` in both operators; or
- validate their dtype at construction and raise a clear Python error documenting the required type.

Add coverage for default integer arrays with JAX x64 enabled for both `RFIVisOp` and `RFIDelayVisOp`.
